### PR TITLE
Implement FIP-0031

### PR DESCRIPTION
**Pull request by arajasek**
Merge strategy: Land #8293 in master, then land this in the appropriate branch (see @jennijuju note below).

This PR implements [FIP-0031](https://github.com/filecoin-project/FIPs/blob/master/FIPS/fip-0031.md). The key changes are:

- [x] Introducing v8 specs-actors and the necessary migration
- [ ] Integrating v8 builtin actors into the FFI (doesn't need the actors to be code-complete) 
- [ ] Supporting v16 in the FVM (this should just be a matter of adding v16 to the supported versions in the FVM)
- [ ] Using content-addressed CIDs in specs-actors (see https://github.com/filecoin-project/specs-actors/issues/1581, https://github.com/filecoin-project/lotus/issues/8066)

Fixes https://github.com/filecoin-project/lotus/issues/8117.
Fixes https://github.com/filecoin-project/lotus/issues/8061.

### CODE REVIEW

**Comment by jennijuju** [comment]
resolves #8061

**Comment by arajasek** [comment]
The test failures are because they try to run with the latest network version (set to v16 in this PR), which is currently unsupported by the FVM and the actors. We could:

- temporarily bump them to using nv15 (this would require undoing some codegen, which is undesirable)
- make getting v8 actors and FVM v16 a blocker for this PR: this probably makes more sense in terms of ordering, but if we decide to do that, I would want to merge `feat/fvm` (the destination branch of this PR) into `master` now (and this can land later).

**Comment by arajasek** [comment]
Closes #8064

**Comment by jennijuju** [comment]
after feat/fvm is merged into master (tmr) then that branch would be closed.
id suggest we then create a nv16 branch, base of release/v1.15.1 and land this PR there

File: chain/vm/mkactor.go [hunk 1]
```diff
@@ -133,6 +134,8 @@ func newAccountActor(ver actors.Version) *types.Actor {
 		code = builtin6.AccountActorCodeID
 	case actors.Version7:
 		code = builtin7.AccountActorCodeID
+	case actors.Version8:
+		code = builtin8.AccountActorCodeID
```
Comment (magik6k) [inline chain/vm/mkactor.go:138]: We'll need to set those code CIDs to real code CIDs

**Review by magik6k — COMMENTED** [review]
Nothing unexpected here, obviously still need real code CIDs

**Comment by raulk** [comment]
Superseded by #8429.

**Pull request closed without merging**